### PR TITLE
[KO] Translation of adversarial_fgsm.ipynb

### PR DESCRIPTION
**Pull request by jaketae**
Uploaded translation of tutorials/generative/adversarial_fgsm.ipynb

### CODE REVIEW

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "yJ-t14sot8iX"
+      },
+      "source": [
+        "Note: 이 문서는 텐서플로 커뮤니티에서 번역했습니다. 커뮤니티 번역 활동의 특성상 정확한 번역과 최신 내용을 반영하기 위해 노력함에도\n",
+        "불구하고 [공식 영문 문서](https://www.tensorflow.org/?hl=en)의 내용과 일치하지 않을 수 있습니다.\n",
+        "이 번역에 개선할 부분이 있다면\n",
+        "[tensorflow/docs](https://github.com/tensorflow/docs) 깃헙 저장소로 풀 리퀘스트를 보내주시기 바랍니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:71]: tensorflow/docs --> tensorflow/docs-l10n
:)

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "W1L3zJP6pPGD"
+      },
+      "source": [
+        "# 빠른 변화도 부호 공격을 이용한 적대적 예제 생성\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:46]: '빠른 변화도 부호 공격' is hard to understand. It's better not to translate it. 
Please change '빠른 변화도 부호 공격' to FGSM in this notebook.

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+      "metadata": {
+        "colab_type": "text",
+        "id": "8dn1-g8BpPDx"
+      },
+      "source": [
+        "이 튜토리얼에서는 Ian Goodfellow *et al*의 [Explaining and Harnessing Adversarial Examples](https://arxiv.org/abs/1412.6572)에 기술된 빠른 변화도 부호 공격(Fast Gradient Signed Method, FGSM)을 이용해 적대적 예제(adversarial example)를 생성하는 방법에 대해 소개합니다. 빠른 변화도 부호도 공격은 신경망 공격 기술들 중 초기에 발견된 방법이자 가장 유명한 방식 중 하나입니다.\n",
+        "\n",
+        "## 적대적 예제란?\n",
+        "\n",
+        "적대적 예제란 신경망을 혼란시킬 목적으로 만들어진 특수한 인풋으로, 신경망으로 하여금 예제를 잘못 분류하도록 합니다. 비록 인간에게 적대적 예제는 일반 예제와 큰 차이가 없어보이지만, 신경망은 적대적 예제를 올바르게 식별하지 못합니다. 이와 같은 신경망 공격에는 여러 종류가 있는데, 본 튜토리얼에서는 화이트 박스(white box) 공격 기술에 속하는 빠른 변화도 부호 공격을 소개합니다. 화이트 박스 공격이란 공격자가 대상 모델의 모든 파라미터값에 접근할 수 있다는 가정 하에 이루어지는 공격을 일컫습니다. 아래 이미지는 Goodfellow et al에 소개된 가장 유명한 적대적 예제인 판다의 사진입니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:88]: There is no reason not to trasnlate 'input'. :)
Please change '인풋' to '입력' in this notebook.

It's easy to misunderstand '예제' as exercise. As you know, example in ML means a sample or observation. 
So I suggest to change '예제' to '샘플' in this notebook.

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        "\n",
+        "![Adversarial Example](images/adversarial_example.png)\n",
+        "\n",
+        "원본 이미지에 특정한 작은 왜곡을 추가하면 신경망으로 하여금 핀다를 높은 신뢰도로 긴팔 원숭이로 잘못 인식하도록 만들 수 있습니다. 이하 섹션에서는 이 왜곡 추가 과정에 대해 살펴보도록 하겠습니다.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:92]: 핀다 --> 판다
:)

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+        "## 빠른 변화도 부호 공격\n",
+        "빠른 변화도 부호 공격은 신경망의 그래디언트(gradient)를 이용해 적대적 예제를 생성하는 기법입니다. 만약 모델의 인풋이 이미지라면, 인풋 이미지에 대한 손실 함수의 그래디언트를 계산하여 그 손실을 최대화하는 이미지를 생성합니다. 이처럼 새롭게 생성된 이미지를 적대적 이미지(adversarial image)라고 합니다. 이 과정은 다음과 같은 수식으로 정리할 수 있습니다:\n",
+        "$$adv\\_x = x + \\epsilon*\\text{sign}(\\nabla_xJ(\\theta, x, y))$$\n",
+        "\n",
+        "각 기호에 대한 설명은 다음과 같습니다.\n",
+        "\n",
+        "*   adv_x : 적대적 이미지.\n",
+        "*   x : 원본 인풋 이미지.\n",
+        "*   y : 원본 인풋 레이블(label).\n",
+        "*   $\\epsilon$ : 왜곡의 정도를 결정하는 승수.\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:103]: I suggest this to convey the meaning of original sentence well.
'왜곡의 정도를 결정하는 승수' --> '왜곡의 양을 적게 만들기 위해 곱하는 수'

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+        "*   $\\theta$ : 모델의 파라미터.\n",
+        "*   $J$ : 손실 함수.\n",
+        "\n",
+        "여기서 흥미로운 사실은  인풋 이미지에 대한 그래디언트가 사용된다는 점입니다. 이는 손실을 최대화하는 이미지를 생성하는 것이 빠른 변화도 부호 공격의 목적이기 때문입니다. 요약하자면, 적대적 예제는 각 픽셀의 손실에 대한 기여도를 그래디언트를 통해 계산한 후, 그 기여도에 따라 픽셀값에 왜곡을 추가함으로써 생성할 수 있습니다. 각 픽셀의 기여도는 연쇄 법칙(chain rule)을 이용해 그래디언트를 계산하는 것으로 빠르게 파악할 수 있습니다. 이것이 인풋 이미지에 대한 그래디언트가 쓰이는 이유입니다. 또한, 대상 모델은 더 이상 학습하고 있지 않기 때문에  (따라서 신경망의 가중치에 대한 그래디언트는 필요하지 않습니다) 모델의 가중치값은 변하지 않습니다. 빠른 변화도 부호 공격의 궁극적인 목표는 이미 학습을 마친 상태의 모델을 혼란시키는 것입니다.\n",
+        "\n",
+        "이제 사전 학습된 모델을 공격해보도록 하겠습니다. 이 튜토리얼에서 사용될 모델은 [ImageNet](http://www.image-net.org/) 가중치로 훈련된 [MobileNetV2](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/applications/MobileNetV2) 모델입니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:109]: '사전 학습된' --> '사전 훈련된'
'ImageNet 가중치로 훈련된' --> 'ImageNet에서 사전 훈련된'

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+      },
+      "outputs": [],
+      "source": [
+        "from __future__ import absolute_import, division, print_function, unicode_literals\n",
+        "\n",
+        "try:\n",
+        "  # %tensorflow_version 명령은 코랩에서만 사용할 수 있습니다.\n",
+        "  %tensorflow_version 2.x\n",
+        "except Exception:\n",
+        "  pass\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:128]: It's not necessary any more.

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "wiTHY8dqxzx7"
+      },
+      "source": [
+        "사전 학습된 MobileNetV2 모델과 ImageNet의 클라스(class) 이름들을 불러옵니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:144]: '사전 학습된' --> '사전 훈련된', '클라스' --> '클래스'

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+        "colab_type": "code",
+        "id": "nqhk2vYx6Ag0"
+      },
+      "outputs": [],
+      "source": [
+        "pretrained_model = tf.keras.applications.MobileNetV2(include_top=True,\n",
+        "                                                     weights='imagenet')\n",
+        "pretrained_model.trainable = False\n",
+        "\n",
+        "# ImageNet 클라스 레이블\n",
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:161]: 클라스 --> 클래스

File: site/ko/tutorials/generative/adversarial_fgsm.ipynb [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "kElVTbF690CF"
+      },
+      "source": [
+        "## 적대적 이미지 생성하기\n",
+        "\n",
+        "### 빠른 변화도 부호 공격 실행하기\n",
+        "첫번째 단계는 예제 생성을 위해 원본 이미지에 가하게 될 왜곡을 생성하는 것입니다. 앞서 살펴보았듯이, 왜곡을 생성할 때에는 인풋 이미지에 대한 그래디언트를 사용합니다."
```
Comment (rickiepark) [inline site/ko/tutorials/generative/adversarial_fgsm.ipynb:254]: '제공할 수 있도록 합니다' --> '제공합니다' 
:)

**Comment by jaketae** [comment]
Hi @rickiepark, thank you so much for the thoughtful review as always. I've made some edits based on your suggestions and comments. Please take a look and let me know what you think!

**Comment by rickiepark** [comment]
Thanks @jaketae 
LGTM 👍 👍 👍

**Comment by jaketae** [comment]
Thank you so much @rickiepark!

Hi @lamberta, have changes been made to how LGTM labels are added to PRs by any chance? I'm slightly confused because I thought typing LGTM triggered some bot action.

**Comment by jaketae** [comment]
@lamberta Got it, thank you for the confirmation!
